pdm/cli/hooks.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from typing import Any
+
+from pdm import signals
+from pdm.project.core import Project
+
+KNOWN_HOOKS = (
+    "post_init",
+    "pre_build",
+    "post_build",
+    "pre_install",
+    "post_install",
+    "pre_lock",
+    "post_lock",
+)
+
+
+class HookManager:
+    projet: Project
+    skip: list[str]
+
+    def __init__(self, project: Project, skip: list[str] | None = None):
+        self.project = project
+        self.skip = skip or []
+
+    @property
+    def skip_all(self) -> bool:
+        return ":all" in self.skip
+
+    @property
+    def skip_pre(self) -> bool:
+        return ":pre" in self.skip
+
+    @property
+    def skip_post(self) -> bool:
+        return ":post" in self.skip
+
+    def try_emit(self, name: str, **kwargs: Any) -> None:
+        if (
+            not self.skip_all
+            and name not in self.skip
+            and not (self.skip_pre and name.startswith("pre_"))
+            and not (self.skip_post and name.startswith("post_"))
+        ):
```

Comment:
These can also benefit from the `should_skip(name)` method.

Reply:
Indeed, yes it does 👌🏼